src/main/java/io/github/treesitter/jtreesitter/Parser.java

```diff
@@ -400,6 +401,10 @@ public String toString() {
     public static final class Options {
         private final Predicate<State> progressCallback;
 
+        /**
+         * @param progressCallback Called when parsing progress was made. Return {@code true} to cancel parsing,
+         *                         {@code false} to continue parsing.
```

Comment:
I hope I understood the logic in https://github.com/tree-sitter/tree-sitter/blob/37a9ecd5b7d52786d0673611f0b8fc54292b1ac1/lib/src/parser.c#L1553 correctly, and `true` does actually cancel parsing.

That is probably worth documenting, otherwise users might be unsure how this `progressCallback` works, or assume that `true` means "continue parsing".

Reply:
But that `true` means cancelling is maybe a bit confusing (even if that matches the tree-sitter behavior). Maybe it would be better to rename it to `cancellationCallback`.

It seems even `ParserTest` might have gotten this wrong?
https://github.com/tree-sitter/java-tree-sitter/blob/1d0055d1548bce2ec169ab0fb344ea19d44e2225/src/test/java/io/github/treesitter/jtreesitter/ParserTest.java#L163
(maybe that should be `>=` instead of `<=`?)

Reply:
> That is probably worth documenting, otherwise users might be unsure how this `progressCallback` works, or assume that `true` means "continue parsing".

That is what I assumed. :skull: